src/routes/users.js

```diff
@@ -87,4 +86,55 @@ router.get('/', jwtAuth, async (req, res) => {
   }
 });
 
+router
+  .route('/:userID')
+  .all(jwtAuth, (req, res, next) => {
+    User.findById(req.params.userID)
+      .then(user => {
+        if (!user) {
+          return res.status(404).json({ Error: 'User not found' });
+        }
+
+        if (req.user._id !== user._id.toString()) {
+          return res.status(403).json({ message: "This isn't your profile" });
+        }
+        res.user = user;
+        next();
+      })
+      .catch(next);
+  })
+  .patch((req, res) => {
+    const { name, userImage } = req.body;
+    const fieldsToUpdate = { name, userImage };
+
+    User.findById(req.params.userID)
+      .then(user => {
+        for (let field in fieldsToUpdate) {
+          if (fieldsToUpdate[field]) {
+            if (fieldsToUpdate[field] === 'null') {
```

Comment:
I tried to understand here. If fieldsToUpdate[field]==='null', why we want to user[field] to be null? Should user[field] be still the same without any changes?